core/ledger/ledger.go

```diff
@@ -108,32 +147,17 @@ func CreateChannel(ctx context.Context, ledgerClient ledgerPb.ChannelsClient, ch
 func CloseChannel(ctx context.Context, ledgerClient ledgerPb.ChannelsClient, signedChannelState *ledgerPb.SignedChannelState) error {
 	closed, err := ledgerClient.CloseChannel(ctx, signedChannelState)
 	if err != nil {
-		log.Error("channel fail to close:", closed.GetState().Channel)
+		log.Errorf("channel fail to close: ", closed.GetState().Channel, err)
 		return err
 	}
 	return nil
 }
 
-func Sign(key *ecdsa.PrivateKey, channelMessage proto.Message) ([]byte, error) {
+func Sign(key libcrypto.PrivKey, channelMessage proto.Message) ([]byte, error) {
```

Comment:
Why not take `*libcrypto.PrivKey`?

Reply:
because I return key.Sign() here, and key is interface.